File: Day1/Javascript/check_prime.js
const readline = require('readline');

function isPrime(n){
  var divisor = 2;

  while (n > divisor){
    if(n % divisor == 0){
     return false; 
    }
    else
      divisor++;
  }
  return true;
}

function main() {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  rl.question("Enter a number: ", function(input) {
    // Convert the input to an integer
    input = parseInt(input);

    // Call the isPrime function to check if the number is prime
    if (isPrime(input)) {
      console.log(input + " is a prime number.");
    } else {
      console.log(input + " is not a prime number.");
    }

    rl.close();
  });
}

// Call the main function
main();

File: Day1/Rust/check_prime.rs
use std::io;

fn is_prime(n: u32) -> bool {
    let mut divisor = 2;

    while n > divisor {
        if n % divisor == 0 {
            return false;
        } else {
            divisor += 1;
        }
    }
    true
}

fn main() {
    println!("Enter a number:");

    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read input.");

    let input: u32 = match input.trim().parse() {
        Ok(num) => num,
        Err(_) => {
            println!("Invalid input. Please enter a valid number.");
            return;
        }
    };

    if is_prime(input) {
        println!("{} is a prime number.", input);
    } else {
        println!("{} is not a prime number.", input);
    }
}